### ISSUE TITLE
regression: Colorblindly now requires global read/write

### ISSUE THREAD

**Issue by wchargin** [oftheheadland/Colorblindly#15]
A recent update (perhaps #14) changed Colorblindly’s permissions to
require reading and changing all data on all websites that I visit.

Just wanted to let you know, in case this wasn’t intentional. I was
impressed to originally see that Colorblindly didn’t require these
permissions, but I’ve had to uninstall it as of the latest update.

**Comment by oftheheadland** [comment]
I agree with you @wchargin, in #17 this is addressed and the current tab functionality is restored

**Comment by wchargin** [comment]
Great. Thank you! I'll check back for a new extension release shortly.

**Comment by oftheheadland** [comment]
Should be live! Thank you.

**Comment by wchargin** [comment]
Confirmed, and reinstalled. Thanks!

**Comment by garethellis36** [comment]
Hello

Sorry to raise this issue from the dead. With the latest update, the extension is requesting global read/write permissions. Is this intentional?

![image](https://github.com/oftheheadland/Colorblindly/assets/6451455/d628190f-a6df-4f80-893f-97d818ce58f8)

**Comment by oftheheadland** [comment]
@garethellis36 From my understanding, as of Chrome Version 121.0.6167.86 (Official Build) (64-bit) and with Chrome Manifest v3 https://developer.chrome.com/docs/extensions/develop/migrate, this permission is required to inject JavaScript into pages.

Anyone please correct me if I'm wrong or if there is a better way to do this with less permissions! This is a small hobby project from when I was a student, I have not kept up with best practices for Chrome Extensions. Thanks!
